tests/Rule/Base/SerializableRuleTestTrait.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Validator\Tests\Rule\Base;
+
+use Yiisoft\Validator\SerializableRuleInterface;
+
+trait SerializableRuleTestTrait
+{
+    abstract public function dataOptions(): array;
+
+    public function beforeTestOptions(): void
```

Comment:
Not crucial, especially in tests, but could be private.

Reply:
done

Reply:
Seems like there are no changes (at least for this method).

Reply:
Oh, fixed.

Reply:
> but could be private

Protected obviously, since it's overridden. :slightly_smiling_face: